aep/general/0005/aep.md

```diff
@@ -0,0 +1,84 @@
+# Designing an API
+
+This AEP serves as a high-level guide on designing an API that adheres to the
+AEPs.
+
+## Process summary
+
+1. enumerate the user case would like your API to satisfy.
+1. identify resources and add them to your API.
+1. identify standard operations and add them to your API.
+1. identify custom operations and add them to your API.
+
+## Enumerate use cases
+
+The first step in designing an API is understanding precisely what you would
+like your user to be able to accomplish your API.
+
+Enumerate one or more use cases to clarify what precisel
+
+For example:
+
+- Create a VM.
+- List all VMs owned by a company.
+- List all VMs owned by a user.
+- Restart a running VM.
+- Trade an in-game item with another player.
+
+Some best practices:
```

Comment:
Nit:  A list of one thing probably doesn't need to be a list...

Reply:
I guess we could switch it later, but I'm hoping someone has more to add! 

Before I switch it back: any best practices we should include?

Reply:
Maybe something about trying to plan for known or likely future use cases, rather than just the highest priority or MVP use cases?

Reply:
Added! PTAL.